### DISCUSSION TITLE
REST store class?

### DISCUSSION THREAD

**Discussion by kathryncrouch** [GMOD/jbrowse-components#1788] [Q&A] [answered]
I'm wondering if JB2 has a REST store class? We are currently making heavy use of the REST store class in JB1. I can't see any documentation about it in JB2. Sorry if I missed it!

Thanks!

**Comment by rbuels** [comment, marked as answer]
We haven't made one yet, but it would not be very hard to do. Initially, would you prefer a data adapter that is 100% compatible with the JBrowse 1 REST API, or would you prefer one that uses a new API, but is more powerful and flexible?

**Reply by kathryncrouch** [reply]
Thanks, glad I wasn't missing something obvious! I am thinking about this in the context of VEuPathDB - we are quite a big team and I certainly can't speak for everyone.

My initial thought is that JB1 compatibility would make migration easier, but we would obviously have to change other things if we were to migrate anyway so I don't think a new API would be a show stopper if it brings benefits.

There are other things preventing us migrating for the time being anyway (notably Apollo, which is re-using our JB1 configurations) so it's not urgent - but I've been playing with a local JB2 instance for a separate project which is really nice and I'm trying to get the migration conversation started!

**Reply by Bjoernsen** [reply]
The missing REST adapter prevents us from switching from JB1 to 2, too. It must not be the same, we can implement new API endpoints for JB2.

**Reply by cmdcolin** [reply]
Thanks for posting interest @Bjoernsen an issue to track this is here if you want to subscribe to it https://github.com/GMOD/jbrowse-components/issues/1825

An alternative is creating a custom plugin to adapter to your backend REST API. I actually like doing this because you have a lot of control over how the client side accesses your REST API instead of molding to what we prescribe. however, hopefully we can develop something similar to JBrowse 1 built in REST API too
